service/Abstractions/DataFormats/IContentDecoder.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.KernelMemory.Pipeline;
+
+namespace Microsoft.KernelMemory.DataFormats;
+
+public interface IContentDecoder
+{
+    IEnumerable<string> SupportedMimeTypes { get; }
+
+    Task<FileContent?> ExtractContentAsync(string handlerStepName, DataPipeline.FileDetails file, string filename, CancellationToken cancellationToken = default);
```

Comment:
It is necessary so that the **TextExtractionHandler** knows that the file has been skipped.

Reply:
what about using a specific exception instead? a nullable object allows to define only a single scenario (skipped in this case), but with exceptions we would allow defining also reasons for skipping and other scenarios, e.g. file too big, timeouts, etc.

Reply:
Initially I was thinking about exceptions, but I have tried to avoid them for performance reasons. But we could definitely use them. In fact, using exceptions we could also solve the logging problems described in the later comments.